Gems/Atom/Feature/Common/Code/Source/CoreLights/SimpleSpotLightFeatureProcessor.cpp

```diff
@@ -303,7 +362,23 @@ namespace AZ
         void SimpleSpotLightFeatureProcessor::UpdateBounds(LightHandle handle)
         {
             const SimpleSpotLightData& data = m_lightData.GetData<0>(handle.GetIndex());
-            m_lightData.GetData<1>(handle.GetIndex()) = SpotLightUtils::BuildBounds(data);            
+            m_lightData.GetData<1>(handle.GetIndex()).m_boundsVariant = SpotLightUtils::BuildBounds(data);
+        }
+
+        void SimpleSpotLightFeatureProcessor::UpdateViewProjectionMatrix(LightHandle handle)
+        {
+            auto& lightData = m_lightData.GetData<0>(handle.GetIndex());
+
+            AZ::Matrix4x4 viewToClipMatrix;
+            float halfFov = acosf(lightData.m_cosOuterConeAngle);
+            float attenuationRadius = LightCommon::GetRadiusFromInvRadiusSquared(lightData.m_invAttenuationRadiusSquared);
+            attenuationRadius = AZStd::max(0.02f, attenuationRadius);
+            MakePerspectiveFovMatrixRH(viewToClipMatrix, halfFov * 2, 1.0f, 0.01f, attenuationRadius, false);
```

Comment:
nit: I get nervous when I see a float multiplied by an integer, too many issues in the past with unintended casting 
```suggestion
            MakePerspectiveFovMatrixRH(viewToClipMatrix, halfFov * 2.0f, 1.0f, 0.01f, attenuationRadius, false);
```